forum/users/serializers.py

```diff
@@ -0,0 +1,42 @@
+from rest_framework import serializers
+from .models import User, Role
+
+class RoleSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Role
+        fields = ('role_id', 'role_name')
+
+class RegisterUserSerializer(serializers.ModelSerializer):
+    roles = RoleSerializer(many=True)
+    password = serializers.CharField(write_only=True)
+
+    class Meta:
+        model = User
+        fields = ('user_id', 'user_name', 'first_name', 'last_name', 'email', 'user_phone', 'roles', 'password', 'created_at', 'updated_at')
+        read_only_fields = ('created_at', 'updated_at')
+
+    def create(self, validated_data):
+        # Вилучаємо дані про ролі та пароль
+        roles_data = validated_data.pop('roles', [])
+        password = validated_data.pop('password')
+        
+       
+        user = User.objects.create(**validated_data)
+        user.set_password(password)
```

Comment:
`set_password` method already performs validation for password hashing, so there’s no need to repeat password setting logic inside the serializer.